nixos/tests/installed-tests/default.nix

```diff
@@ -0,0 +1,68 @@
+# NixOS tests for gnome-desktop-testing-runner using software
+
+{ system ? builtins.currentSystem,
+  config ? {},
+  pkgs ? import ../../.. { inherit system config; }
+}:
+
+with import ../../lib/testing-python.nix { inherit system pkgs; };
+with pkgs.lib;
+
+let
+
+  callInstalledTest = pkgs.newScope { inherit makeInstalledTest; };
+
+  makeInstalledTest =
+    { tested               # Package to test. Needs to have an installedTests output
+    , testConfig ? {}      # Config to inject into machine
+    , extraTestScript ? "" # Test script snippet to inject before gnome-desktop-testing-runnner begins
+    , withX11 ? false      # Does test need X11?
+    , testRunnerFlags ? "" # Extra flags to pass to gnome-desktop-testing-runner
+    }:
+    makeTest rec {
+      name = tested.name;
+
+      meta = {
+        maintainers = tested.meta.maintainers;
+      };
+
+      machine = { ... }: {
+        imports = [
+          testConfig
+        ] ++ optional withX11 ../common/x11.nix;
+
+        environment.systemPackages = with pkgs; [ gnome-desktop-testing ];
+
+      };
+
+      testScript =
+        optionalString withX11
```

Comment:
Could we get the opening quotes on the same line as `optionalString` as is customary with arrays.